erpnext/accounts/doctype/repost_payment_ledger/repost_payment_ledger.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2022, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import copy
+
+import frappe
+from frappe import _, qb
+from frappe.model.document import Document
+from frappe.query_builder.custom import ConstantColumn
+from frappe.utils.background_jobs import is_job_queued
+
+from erpnext.accounts.utils import _delete_pl_entries, create_payment_ledger_entry
+
+VOUCHER_TYPES = ["Sales Invoice", "Purchase Invoice", "Payment Entry", "Journal Entry"]
+
+
+def repost_ple_for_voucher(voucher_type, voucher_no, gle_map=None):
+	if voucher_type and voucher_no and gle_map:
+		_delete_pl_entries(voucher_type, voucher_no)
+		create_payment_ledger_entry(gle_map, cancel=0)
+
+
+@frappe.whitelist()
+def start_payment_ledger_repost(docname=None):
+	"""
+	Repost Payment Ledger Entries for Vouchers through Background Job
+	"""
+	if docname:
+		repost_doc = frappe.get_doc("Repost Payment Ledger", docname)
+		if repost_doc.docstatus == 1 and repost_doc.repost_status == "Queued":
+
+			for entry in repost_doc.repost_vouchers:
+				doc = frappe.get_doc(entry.voucher_type, entry.voucher_no)
+
+				if doc.doctype in ["Payment Entry", "Journal Entry"]:
+					gle_map = doc.build_gl_map()
+				else:
+					gle_map = doc.get_gl_entries()
+
+				repost_ple_for_voucher(entry.voucher_type, entry.voucher_no, gle_map)
+
+			frappe.db.set_value(repost_doc.doctype, repost_doc.name, "repost_status", "Completed")
+
+
+class RepostPaymentLedger(Document):
+	def __init__(self, *args, **kwargs):
+		super(RepostPaymentLedger, self).__init__(*args, **kwargs)
+		self.vouchers = []
+
+	def validate(self):
+		if not self.company:
+			frappe.throw(_("Company is Mandatory"))
```

Comment:
Needs checking again?
Both of these seem to have `reqd` as 1

Reply:
Removed